azure-pipelines.yml

```diff
@@ -1,813 +1,223 @@
-            set -e
-            cd qiskit-tutorials
-            sphinx-build -b html . _build/html
-          env:
-            QISKIT_PARALLEL: False
-        - task: ArchiveFiles@2
-          inputs:
-            rootFolderOrFile: 'qiskit-tutorials/_build/html'
-            archiveType: tar
-            archiveFile: '$(Build.ArtifactStagingDirectory)/html_tutorials.tar.gz'
-            verbose: true
-        - task: PublishBuildArtifacts@1
-          displayName: 'Publish docs'
-          inputs:
-            pathtoPublish: '$(Build.ArtifactStagingDirectory)'
-            artifactName: 'html_tutorials'
-            Parallel: true
-            ParallelCount: 8
+      jobs:
+        - job: "Comment"
+          steps:
+            - task: GitHubComment@0
+              inputs:
+                gitHubConnection: Qiskit
+                repositoryName: ${{ variables['Build.Repository.Name'] }}
+                id: 7864
+                comment: Nightly test job failed at commit $(Build.SourceVersion).  View the logs at $(System.TeamFoundationCollectionUri)$(System.TeamProject)/_build/results?buildId=$(Build.BuildId).
+
+  # Full PR suite.
+  - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+    # The preliminary stage should be small in both total runtime (including
+    # provisioning) and resources required.  About half of PR commits result in
+    # a CI failure, and over 90% of these are in linting, documention or a test
+    # failure that would affect _any_ OS or Python version.  The goal in the
+    # first stage is to catch the vast majority of failures with minimal cost.
+    - stage: "Lint_Docs_Prelim_Tests"
+      displayName: "Preliminary tests"
+      jobs:
+        - template: ".azure/lint-linux.yml"
+          parameters:
+            pythonVersion: ${{ parameters.minimumPythonVersion }}
+
+        - template: ".azure/docs-linux.yml"
+          parameters:
+            pythonVersion: ${{ parameters.minimumPythonVersion }}
+
+        - template: ".azure/test-linux.yml"
+          parameters:
+            pythonVersion: ${{ parameters.minimumPythonVersion }}
+            testQPY: false
+            testImages: false
+
+    # The rest of the PR pipeline is to test the oldest and newest supported
+    # versions of Python, along with the integration tests (via the tutorials).
+    # It's very rare for a failure to be specific to an intermediate version of
+    # Python, so we just catch those in the cron-job pipeline to reduce the
+    # amount of resources used.
+    - stage: "Tutorials_and_Tests"
+      displayName: "Main tests"
+      dependsOn: "Lint_Docs_Prelim_Tests"
+      jobs:
+        - template: ".azure/tutorials-linux.yml"
+          parameters:
+            pythonVersion: ${{ parameters.tutorialsPythonVersion }}
+
+        - template: ".azure/test-linux.yml"
+          parameters:
+            pythonVersion: ${{ parameters.maximumPythonVersion }}
+            testQPY: true
+            testImages: true
```

Comment:
Yeah - the reason was to shift the QPY and image tests into the second stage rather than the first.  Happy to move it back, but perhaps it's best just to leave this comment visible on the PR for the history in case we need to refer back to it.

Reply:
Yeah, I figured that was the reason. This comment was more just me playing catch the difference in the configuration